src/core/config/descriptors/parameter_descriptor.cpp

```diff
@@ -7,6 +7,120 @@
+    if (*str == '-') {
+        return false;
+    }
+
+    char *endptr;
+    errno = 0; // Reset errno before the call.
+    unsigned long long value = strtoull(str, &endptr, 10);
+
+    // Check for parsing errors from strtoull.
+    if (endptr == str || errno == ERANGE) {
+        return false;
+    }
+
+    // Skip whitespace between the number and the unit.
+    while (isspace(static_cast<unsigned char>(*endptr))) {
+        endptr++;
+    }
+
+    const char *unit_start = endptr;
+
+    // The remainder of the string is the unit. Find its end.
+    while (*endptr != '\0' && !isspace(static_cast<unsigned char>(*endptr))) {
+        endptr++;
+    }
+
+    // Check for any non-whitespace trailing characters.
+    char *temp = endptr;
+    while (isspace(static_cast<unsigned char>(*temp))) {
+        temp++;
+    }
+    if (*temp != '\0') {
+        return false; // Disallow trailing characters.
+    }
+
+    uint64_t multiplier = 1; // Default to bytes.
+    std::string unit(unit_start, endptr - unit_start);
+
+    if (!unit.empty()) {
+        if (strcasecmp(unit.c_str(), "B") == 0) {
+            multiplier = 1;
+        } else if (strcasecmp(unit.c_str(), "K") == 0 || strcasecmp(unit.c_str(), "KB") == 0) {
+            multiplier = 1024ULL;
+        } else if (strcasecmp(unit.c_str(), "M") == 0 || strcasecmp(unit.c_str(), "MB") == 0) {
+            multiplier = 1024ULL * 1024ULL;
+        } else if (strcasecmp(unit.c_str(), "G") == 0 || strcasecmp(unit.c_str(), "GB") == 0) {
+            multiplier = 1024ULL * 1024ULL * 1024ULL;
+        } else {
+            return false; // Invalid unit specified.
+        }
+    }
+
+    // Check for overflow before multiplication.
+    if (value > 0 && multiplier > 1 && value > ULLONG_MAX / multiplier) {
+        throw_xlio_exception("Memory value '" + std::string(str) +
+                             "' is too large and would cause an overflow.");
+    }
+
+    size_in_bytes = value * multiplier;
+    return true;
+}
+
+value_transformer_t parameter_descriptor::create_memory_size_transformer()
+{
+    return [](const std::experimental::any &val) -> std::experimental::any {
+        // Only transform string values
+        if (val.type() == typeid(std::string)) {
+            std::string str_val = std::experimental::any_cast<std::string>(val);
+            uint64_t parsed_size;
+
+            if (parameter_descriptor::parse_memory_size(str_val.c_str(), parsed_size)) {
+                // Check if the parsed value exceeds the max value for int64_t
+                if (parsed_size > static_cast<uint64_t>(LLONG_MAX)) {
+                    throw_xlio_exception("Memory value '" + str_val +
+                                         "' exceeds maximum supported size.");
+                }
+                return static_cast<int64_t>(parsed_size);
+            }
+        }
+
+        // Return original value if not a string that can be transformed
+        return val;
```

Comment:
If the value cannot be transformed its probably an error that should be displayed to the user. Would this any value propagate to an error?